### PR TITLE
Add filter to notes datastore where clauses

### PR DESCRIPTION
**Pull request by rrennick**
See #2714

Add filter to notes datastore where clauses

### Detailed test instructions:

- Load notes without PR
- Go to regular WP Admin page
- Clear transients
- Switch to PR
- Verify same notes are shown

### Changelog Note:

Dev: Add filter to notes datastore where clauses.

### CODE REVIEW

**Comment by rrennick** [comment]
@psealock Can you have a quick look at this and comfirm this is the type of filters required for the issue? Should the value(s) being returned by the filter be `sql_esc`aped?

**Comment by jeffstieler** [comment]
> @jeffstieler do you know if `esc_sql` followed by a `$wpdb->prepare` is sufficient to sanitize an added `where` clause?

We won't be able to handle all cases - literally any value can come back from the filter. Even `WP_Query` doesn't try :) - look for `apply_filters_ref_array` in `class-wp-query.php`.

**Comment by jeffstieler** [comment]
> @jeffstieler do you know if `esc_sql` followed by a `$wpdb->prepare` is sufficient to sanitize an added `where` clause?

We won't be able to handle all cases - literally any value can come back from the filter. Even `WP_Query` doesn't try :) - look for `apply_filters_ref_array` in `class-wp-query.php`.

File: src/Notes/DataStore.php [hunk 1]
```diff
@@ -371,7 +372,7 @@ public function get_notes_where_clauses( $args = array() ) {
 			$where_clauses .= " AND status IN ($escaped_status_types)";
 		}
 
-		return $where_clauses;
+		return apply_filters( 'wc_admin_notes_where_clauses', $where_clauses, $args );
```
[the remark this answers is not in the document]
Comment (rrennick) [reply src/Notes/DataStore.php]: Added in https://github.com/woocommerce/woocommerce-admin/pull/2819/commits/1f00f302e9b3f1e7d36a6c4af3d878fd5840212f

**Pull request merged**